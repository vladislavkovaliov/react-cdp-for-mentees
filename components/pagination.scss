@import '../constants/_variables.scss';

ul.pagination {
  display: flex;
  justify-content: center;
  list-style: none;

  & li {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 4px 8px;
    margin: 14px 4px 14px 4px;
    border: 1px solid #000000;
    border-radius: 8px;
    height: 18px;

    @media screen and (max-width: 368px) {
      &.break {
        display: none;
      }
    }

    a {
      cursor: pointer;
      outline: none;
    }

    &.break {
      border: none;
    }

    &.selected {
      border-color: $red;
    }
  }
}
