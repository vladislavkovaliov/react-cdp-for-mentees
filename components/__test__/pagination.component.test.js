import React from 'react';
import { Pagination, mapStateToProps, mapDispatchToProps } from '../pagination.component';
import { updateViewport } from '../../actions/viewport.action';
import withRedux from 'next-redux-wrapper';
import { makeStore } from '../../utils/create-store.util';

import shallow from 'enzyme/shallow';

jest.mock('../../actions/viewport.action', () => ({
  updateViewport: jest.fn(),
}));

jest.mock('../../utils/create-store.util', () => ({
  makeStore: jest.fn(),
}));

jest.mock('next-redux-wrapper', () => {
  return jest.fn(() => jest.fn());
});

describe('<Pagination />', () => {
  describe('connect()', () => {
    test('should call connect with mapStateToProps and mapDispatchToProps', () => {
      expect(withRedux).toHaveBeenCalledWith(makeStore, mapStateToProps, mapDispatchToProps);
    });
  });

  describe('mapStateToProps', () => {
    test('should return object with movies', () => {
      const window = {
        isMobile: false,
      };
      const result = mapStateToProps({ window });

      expect(result).toHaveProperty('window', window);
    });
  });

  describe('mapDispatchToProps', () => {
    test('should call searchMovies action', () => {
      const result = mapDispatchToProps(null /* dispatch */);

      result.updateViewport({});

      expect(updateViewport).toHaveBeenCalled();
    });

    test('should return object with method updateViewport', () => {
      const result = mapDispatchToProps(null /* dispatch */);

      expect(result).toHaveProperty('updateViewport');
    });
  });

  describe('Shaphot', () => {
    test('Shapshot <Pagination /> via desktop', () => {
      const wrapper = shallow(
        <Pagination
          updateViewport={updateViewport}
          window={{ isMobile: false }}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    test('Shapshot <Pagination /> via mobile', () => {
      const wrapper = shallow(
        <Pagination
          updateViewport={updateViewport}
          window={{ isMobile: true }}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
