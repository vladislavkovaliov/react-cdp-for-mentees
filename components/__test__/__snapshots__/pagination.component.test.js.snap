// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> Shaphot Shapshot <Pagination /> via desktop 1`] = `
<PaginationBoxView
  activeClassName="selected"
  breakLabel="..."
  containerClassName="pagination"
  disableInitialCallback={false}
  disabledClassName="disabled"
  marginPagesDisplayed={1}
  nextClassName="next"
  nextLabel="Next"
  pageCount={10}
  pageRangeDisplayed={3}
  previousClassName="previous"
  previousLabel="Previous"
  updateViewport={
    [MockFunction] {
      "calls": Array [
        Array [
          null,
          Object {},
        ],
      ],
    }
  }
  window={
    Object {
      "isMobile": false,
    }
  }
/>
`;

exports[`<Pagination /> Shaphot Shapshot <Pagination /> via mobile 1`] = `
<PaginationBoxView
  activeClassName="selected"
  breakLabel="..."
  containerClassName="pagination"
  disableInitialCallback={false}
  disabledClassName="disabled"
  marginPagesDisplayed={0}
  nextClassName="next"
  nextLabel="Next"
  pageCount={10}
  pageRangeDisplayed={0}
  previousClassName="previous"
  previousLabel="Previous"
  updateViewport={
    [MockFunction] {
      "calls": Array [
        Array [
          null,
          Object {},
        ],
      ],
    }
  }
  window={
    Object {
      "isMobile": true,
    }
  }
/>
`;
